### PR TITLE
Fix parse create database

### PR DESCRIPTION
**Pull request by XisiHuang**
issue #1818 becasue the second optional elements B satisfy createdb_opt_name:IDENT, and createdb_opt_item return nil,.

### CODE REVIEW

**Comment by tamird** [comment]
Can you add some tests?

File: sql/parser/sql.y [hunk 1]
```diff
@@ -1575,8 +1575,7 @@ createdb_opt_item:
 // option will silently break. Best defense is to provide a regression test
 // exercising every such option, at least at the syntax level.
 createdb_opt_name:
-  IDENT {}
-| CONNECTION LIMIT {}
+  CONNECTION LIMIT {}
```
Comment (petermattis) [inline sql/parser/sql.y]: I'd rather just get rid of the `createdb_opt_with` and `createdb_opt_list` rules completely. Neither are being used to create nodes in the syntax tree.

Comment (XisiHuang) [inline sql/parser/sql.y]: Options of "CREATE DATABASE" are supposed in many popular DB  in the market, maybe it's necessary to reserve for user to substitute esaily in the future.

Cocoroach has no plan to implement it currently?  even later?

Comment (petermattis) [inline sql/parser/sql.y]: Yes, we'll almost certainly support options for `CREATE DATABASE` at some point, but the options here are all postgres specific. Re-adding the grammar rules for parsing the cockroach specific options will not be difficult.

Comment (XisiHuang) [inline sql/parser/sql.y]: Got it,i'll open another another PR for remove the options completely if this one can pass. thanks. :)

**Pull request merged**